apps/wallet/src/ui/app/redux/slices/transaction-requests/index.ts

```diff
@@ -65,10 +66,105 @@ export const loadTransactionResponseMetadata = createAsyncThunk<
     }
 );
 
+export type TxnMetaResponse = {
+    coinSymbol?: string | null;
+    amount?: number | null;
+    objectId?: string;
+    isListing?: boolean;
+} | null;
+
+const getRequestCost = (
+    txEffects: TransactionEffects,
+    address: string
+): TxnMetaResponse | null => {
+    const events = txEffects?.events || [];
+
+    let sumCoin = 0;
+    const coinMeta: {
+        objectId?: string;
+        amount?: number;
+        coinSymbol?: string;
+        isListing?: boolean;
+    } = {};
+
+    for (const event of events) {
+        if (
+            'coinBalanceChange' in event ||
+            'transferObject' in event ||
+            'moveEvent' in event
+        ) {
+            // Get the amount in move events
+            if (
+                event.moveEvent?.sender === address &&
+                event.moveEvent?.fields?.price
+            ) {
+                sumCoin += event.moveEvent.fields.price;
+                coinMeta.amount = sumCoin;
+                coinMeta.coinSymbol = '0x2::sui::SUI';
```

Comment:
What's a moveEvent? how do we know that price is SUI?

Reply:
I assume the listing is in SUI, do we have any instances where it is not

Reply:
It seems that this function in general is very specific to a few use cases `getRequestCost` and it feels that we can display wrong data in case we get different type of events

Reply:
Only `coinBalanceChange`, `transferObject`, `moveEvent` matter, other event types are ignored.